apps/ledger-live-mobile/src/components/ValidateMessageOnDevice.tsx

```diff
@@ -30,9 +39,45 @@ export default function ValidateOnDevice({ device, message, account }: Props) {
       color: colors.text,
     },
   ];
+
+  const mainAccount = getMainAccount(account, null);
+  const {
+    message,
+    fields,
+  }: {
+    message?: string | null | undefined;
```

Comment:
You're right here, I'll remove the undefined. 👍 

Concerning the `key: null | undefined` vs `key?: null`, it's actually not the same. Having a key in an object and not having it is not representing the same object. 

As an example of a real usage where it would matter to have the property or not:
```typescript
const objectA = { foo: "bar", bar: undefined };
const objectB = { foo: "bar" };
const objectC = { foo: "bar", bar: "foo" }

const compareObjectByKey = (o1, o2) => {
  let isSame = true;

  for (const key of Object.keys(o1)) {
    isSame = Boolean(o2[key] && o1[key] === o2[key]);
    if (!isSame) break;
  }

  return isSame;
}

compareObjectByKey(objectA, objectC) // false
compareObjectByKey(objectB, objectC) // true
```

And that's the kind of things we do very often in the live to update objects only when necessary to prevent re-rendering.